### DISCUSSION TITLE
Isn't this a bug when applying text format?

### DISCUSSION THREAD

**Discussion by tmarplatt** [vkbo/novelWriter#1323] [General]
Currently using 2.0.3. When you type a word and then apply a text format (say, _italics_) the markup is applied as expected. Keep in mind that the cursor is at the end of the word, like so (illustrated with a pipe |).

`This is some emphasized word|`

becomes

`This is some emphasized _word|_`

But is this correct? Wouldn't a writer expect the cursor to be pushed out of the markup, so that they can keep typing outside the formatted word?

You could argue that novelWriter has no real way of knowing what the user intends to do. But then again text is not being selected, instead the cursor is at a word boundary. It makes a lot of sense and follows intuition that the user is only applying format to the afore-written word, and that whatever is getting written next will remain unformatted.

Unfortunately this is also how markup editors work, such as this one at Github I'm posting from. But shouldn't novel-writing software stick to other user expectations (i.e. those of writers, not programmers)?

Should this be considered a bug, I'll gladly open an issue about it.

**Comment by vkbo** [comment]
This is the optional feature that assumes that when you apply formatting without selecting anything, the formatting applies to the word under the cursor.

I agree that in your example it makes sense to move the cursor to after the formatting. However, the idea for the current behaviour is to preserve the ability to toggle formatting on and off for the word, Like for instance if you wanted italics, but hit bold, or perhaps wanted both. In that case, you would have to move the cursor again.

As you say, it's hard to guess the intention of the user, but testing a bit in the editor, I can probably make the feature a little smarter. Feel free to open an issue. This is definitely not a bug though, rather a feature request.

**Reply by tmarplatt** [reply]
(Moved here)

I've submitted issue #1333.

Incidentally, here's a suggestion for a new tag for issues: Paper-cut. Small bugs or feature requests that, unsolved, worsen user experience at marginal, sometimes niche detriment, but that as a whole become a sizable hindrance. Fixing them adds a degree of polish that is satisfying to discover in software.

**Reply by vkbo** [reply]
Thanks!

I recently added a label called "usability" for exactly that purpose, although "paper-cut" has a much more on-topic ring to it 😃 

The label is orange, which is not exactly a bug (red) but still a hindrance or friction. I use blue for improvements, but the distinction is mostly that improvements can wait till another minor release, while orange ones can be included in patches if the changes are small.
